### PR TITLE
Add plant to garden

### PR DESCRIPTION
**Pull request by drewwilliams5280**
#66 completed

If the plant already exists in the db, plant is easily added.

If the plant does not exist in the db, it is created and added to the garden.

Comments on code below.

This built upon what I did with the previous pull request [here](https://github.com/My-Solar-Garden/rails_backend/pull/65) so if you merge this before that, please feel free to just close that one.

### CODE REVIEW

File: app/controllers/api/v1/gardens/plants_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Api::V1::Gardens::PlantsController < ApplicationController
+  def create
+    garden = Garden.find(params[:id])
+    if Plant.exists?(params[:plant_id])
+      garden.add_plant(Plant.find(params[:plant_id]))
```
Comment (drewwilliams5280) [inline app/controllers/api/v1/gardens/plants_controller.rb:5]: We will need FE to pass the plant id with the params key of :plant_id

File: app/controllers/api/v1/gardens/plants_controller.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    else
+      garden.create_and_add_plant(plant_params)
+    end
+  end
```
Comment (drewwilliams5280) [inline app/controllers/api/v1/gardens/plants_controller.rb:9]: We will need FE to pass the plant id with the params key of :plant_id

File: app/models/garden.rb [hunk 3]
```diff
@@ -7,4 +7,15 @@ class Garden < ApplicationRecord
...
+  def add_plant(plant)
+    GardenPlant.create!(garden: self, plant: plant, planted_date: Date.today)
+  end
+
+  def create_and_add_plant(plant_params)
+    plant = Plant.new(plant_params)
+    if plant.save
+      GardenPlant.create!(garden: self, plant: plant, planted_date: Date.today)
+    end
+  end
```
Comment (drewwilliams5280) [inline app/models/garden.rb:20]: If front end wants to let the user specify which date they planted the plant, we will need them to send with params.

File: config/routes.rb [hunk 4]
```diff
@@ -3,9 +3,15 @@
   namespace :api do
     namespace :v1 do
       get '/garden_healths/search', to: 'search#index'
```
Comment (drewwilliams5280) [inline config/routes.rb:5]: Might need to namespace this.

**Review by dcoleman21 — APPROVED** [review]
Let's just make sure that we add these param key specifications to a FE ticket for plants so that it is not lost in translation and easy to access when calling BE endpoints for FE

**Pull request merged**